Mail/Views/SplitView.swift

```diff
@@ -101,12 +109,33 @@ struct SplitView: View {
                 }
             }
         }
+        .floatingPanel(isPresented: $isShowingUpdateAvailable) {
+            DiscoveryView(item: .updateDiscovery) { /* Empty on purpose */ } completionHandler: { update in
+                guard update else { return }
+                let url: URLConstants = Bundle.main.isRunningInTestFlight ? .testFlight : .appStore
+                openURL(url.url)
+            }
+        }
+        .floatingPanel(isPresented: $isShowingSyncDiscovery, content: {
+            DiscoveryView(item: .syncDiscovery) {
+                UserDefaults.shared.shouldPresentSyncDiscovery = false
+            } completionHandler: { update in
+                guard update else { return }
+                isShowingSyncProfile = true
+            }
+        })
+        .sheet(isPresented: $isShowingSyncProfile) {
+            SyncProfileNavigationView()
+        }
         .sheet(item: $navigationState.editedDraft) { editedDraft in
             ComposeMessageView(editedDraft: editedDraft, mailboxManager: mailboxManager)
         }
         .onChange(of: scenePhase) { newScenePhase in
             guard newScenePhase == .active else { return }
             Task {
+                // We don't want to show both DiscoveryView at the same time
+                isShowingUpdateAvailable = try await VersionChecker.standard.showUpdateVersion()
```

Comment:
Can you move the call `try await VersionChecker.standard.showUpdateVersion()` below `async let _ = try? mailboxManager.refreshAllSignatures()` so it won't block the execution?